test-basic/write-test.js

```diff
@@ -0,0 +1,858 @@
+            done(e);
+        });
+    });
+
+    it('test write with json attachments', function (done) {
+        const bindingParam = "bindingJSONFiles";
+
+        const rows = [{rowId: 1, doc: 'doc1.json', uri: '/test/fromParam/doc1.json'}, {rowId: 1, doc: 'doc2.json', uri: '/test/fromParam/doc2.json'}];
+        // const attachments = [{"doc1.json": "{\"desc\":2}"}];
+        const attachments = [{'doc1.json': {desc: "desc1"}, 'doc2.json': {desc: "desc2"}}];
+        const metadata = {"attachments": {"docs": [{"rowsField": bindingParam, "column": "doc"}]}};
+        const outputCols = [{"column": "rowId", "type": "integer", "nullable": false}, {
+            "column": "doc",
+            "type": "none",
+            "nullable": true
+        }, {
+            "column": "uri",
+            "type": "string",
+            "nullable": true
+        }];
+
+        const planBuilderTemplate = op.fromParam(bindingParam, null, outputCols).write();
+        const bindParam = {[bindingParam]: rows, attachments, metadata};
+
+        db.rows.query(planBuilderTemplate, null, bindParam).then(res => {
+            try {
+                const rows = res.rows;
+                console.log(rows);
+                rows[0].rowId.value.should.equal(1);
+                rows[0].doc.value.should.equal("doc1.json");
+                rows[0].uri.value.should.equal('/test/fromParam/doc1.json');
+                done();
+            } catch (e) {
+                done(e);
+            }
+        }).catch(e => {
+            done(e);
+        });
+    });
+
+    it('test write with json attachments-2', function (done) {
+        const bindingParam = "bindingJSONFiles";
+
+        const rows = [{rowId: 1, doc: 'doc1.json', uri: '/test/fromParam/doc1.json'}];
+        const attachments = [{"doc1.json": "{\"desc\":2}"}];
+        // const attachments = [{'doc1.json': {desc: "desc1"}}];
+        const metadata = {"attachments": {"docs": [{"rowsField": bindingParam, "column": "doc"}]}};
+        const outputCols = [{"column": "rowId", "type": "integer", "nullable": false}, {
+            "column": "doc",
+            "type": "none",
+            "nullable": true
+        }, {
+            "column": "uri",
+            "type": "string",
+            "nullable": true
+        }];
+
+        const planBuilderTemplate = op.fromParam(bindingParam, null, outputCols).write();
+        const bindParam = {[bindingParam]: rows, attachments, metadata};
+
+        db.rows.query(planBuilderTemplate, null, bindParam).then(res => {
+            try {
+                const rows = res.rows;
+                console.log(rows);
+                rows[0].rowId.value.should.equal(1);
+                rows[0].doc.value.should.equal("doc1.json");
+                rows[0].uri.value.should.equal('/test/fromParam/doc1.json');
+                done();
+            } catch (e) {
+                done(e);
+            }
+        }).catch(e => {
+            done(e);
+        });
+    });
```

Comment:
This is the test with 1 attachment, in an array.

`2023-04-20 19:27:20.081 Info: Status 500: OPTIC-INVALARGS: fn.error(null, 'OPTIC-INVALARGS', 'write() - Cannot update doc URI "'+docUri+'" because document does not exist in the database and in memory document not provided.'); -- Invalid arguments: write() - Cannot update doc URI "/test/fromParam/doc1.json" because document does not exist in the database and in memory document not provided.`